From ff396cead32151e3013d6f05124c0602ce8b3e9c Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Thu, 18 Jan 2024 08:15:56 -0600
Subject: [PATCH] Use thin lto

When fat lto is used, the compiler takes too much memory and the OS
kills rustc. Thanks to @mbargull for the tip.
---
 bindings/python/Cargo.toml | 2 +-
 graph/Cargo.toml           | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/bindings/python/Cargo.toml b/bindings/python/Cargo.toml
index 8efa26b4..c241c6c4 100644
--- a/bindings/python/Cargo.toml
+++ b/bindings/python/Cargo.toml
@@ -60,7 +60,7 @@ opt-level = 3
 # time.
 # Perform "fat" LTO which attempts to perform optimizations across all crates 
 # within the dependency graph
-lto = "fat"
+lto = "thin"
 
 # When a crate is split into multiple codegen units, LLVM is able to process 
 # them in parallel. Increasing parallelism may speed up compile times, but may 
diff --git a/graph/Cargo.toml b/graph/Cargo.toml
index bd43fe40..345662b9 100644
--- a/graph/Cargo.toml
+++ b/graph/Cargo.toml
@@ -73,7 +73,7 @@ windows = { version = "0.32.0", features = ["alloc",  "Win32_System_Memory", "Wi
 
 [profile.release]
 opt-level = 3
-lto = "fat"
+lto = "thin"
 overflow-checks = false   # Disable integer overflow checks.
 debug = false            # Include debug info.
 debug-assertions = false  # Enables debug assertions.
-- 
2.42.0

